### PR TITLE
added not_haskell module and tests

### PR DESCRIPTION
**Pull request by pdlla**

### CODE REVIEW

File: src/not_haskell.erl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+-spec mapAccumL(Fun, Acc, List) -> {Acc, List} when
+  %% boo args are backwards... why...
+  Fun :: fun((Elem :: T, Acc) -> {Acc, Out}),
+  Acc :: term(),
+  List :: [T],
+  Out :: term(),
+  T :: term().
+mapAccumL(Fun, Acc, List) ->
+  lists:foldl(fun(_x, {_acc, _outlist}) ->
```
Comment (antonbobkov) [inline src/not_haskell.erl]: not the best to lead variable with _?
in erlang this supresses not used errors

File: src/not_haskell.erl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {NAcc, NOut} = Fun(_x, _acc),
+    %% inefficient snocing, w/e
+    %% TODO implement using foldr instead
```
Comment (antonbobkov) [inline src/not_haskell.erl]: you can also reverse the list first

Comment (pdlla) [reply src/not_haskell.erl]: :o

File: src/not_haskell.erl (hunk 1 above)
Comment (pdlla) [reply src/not_haskell.erl]: :o, that's a useful thing to know >:)

File: src/not_haskell.erl (hunk 2 above)
Comment (pdlla) [reply src/not_haskell.erl]: in haskell foldl' is implemented using foldr by folding a function and evaluating it. foldr' is implemented in the same way using foldl

    foldl' :: (b -> a -> b) -> b -> t a -> b
    foldl' f z0 xs = foldr f' id xs z0
      where f' x k z = k $! f z x

Comment (pdlla) [reply src/not_haskell.erl]: and obv. this is how foldl is implemented :scream: 

 foldl :: (b -> a -> b) -> b -> t a -> b
    foldl f z t = appEndo (getDual (foldMap (Dual . Endo . flip f) t)) z

File: src/not_haskell.erl (hunk 1 above)
Comment (antonbobkov) [reply src/not_haskell.erl]: haha, i saw this in your other code and though it was intentional

Comment (antonbobkov) [reply src/not_haskell.erl]: http://erlang.org/pipermail/erlang-questions/2008-April/034468.html

File: src/not_haskell.erl (hunk 2 above)
Comment (antonbobkov) [reply src/not_haskell.erl]: whats the difference between foldl and foldl'? 😱

**Comment by antonbobkov** [comment]
also, omg, missed opportunity to call this module potato_haskell

**Comment by pdlla** [comment]
:scream: :scream: :scream:  quick, revoke your approval!

File: src/not_haskell.erl (hunk 2 above)
Comment (pdlla) [reply src/not_haskell.erl]: foldl' is strict version of foldl.
The ! after the $ makes the function application strict.

**Comment by antonbobkov** [comment]
haha, you can still rename it in this branch

File: src/not_haskell.erl (hunk 2 above)
Comment (antonbobkov) [reply src/not_haskell.erl]: what's a strict function application? 😱 😱

Comment (pdlla) [reply src/not_haskell.erl]: not lazy :sleepy:

Comment (antonbobkov) [reply src/not_haskell.erl]: neat, didn't realize that was a thing

**Comment by pdlla** [comment]
potato haskell!

**Comment by pdlla** [comment]
how do I say I made the requested changes D:?

**Pull request merged**